### PR TITLE
174 responder 2

### PR DESCRIPTION
**Pull request by yahgwai**
Added the tests for multiresponder

### CODE REVIEW

File: src/responder/gasQueue.ts [hunk 1]
```diff
@@ -288,20 +285,4 @@ export class GasQueue {
...
-    }
-
-    /**
-     * Returns true if a queue item with this identifier exists in this queue
-     * @param identifier
-     */
-    public contains(identifier: PisaTransactionIdentifier) {
-        return this.queueItems.findIndex(i => i.request.identifier.equals(identifier)) !== -1;
-    }
 }
```
Comment (bigspider) [inline src/responder/gasQueue.ts]: The error message in the constructor of GasQueueItem says `"Current gas price must be greater than ideal gas price"` but it should say `"Current gas price must not be smaller than ideal gas price"`

File: test/src/responder/multiResponder.test.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+            errorGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+
+        await responder.startResponse(appointmentId, responseData);
+
+        verify(transactionTrackerMock.addTx(anything(), anything())).never();
+    });
```
Comment (bigspider) [inline test/src/responder/multiResponder.test.ts]: How does the caller of `startResponse` know that the response was not started successfully? Who is responsible for retrying?
For the dedicated responder, events are emitted and the ResponderManager could react accordingly; what is the plan here?

File: src/responder/gasQueue.ts (hunk 1 above)
Comment (yahgwai) [reply src/responder/gasQueue.ts]: Done.

File: test/src/responder/multiResponder.test.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        );
+        await responder.startResponse(appointmentId, responseData);
+        const queueBefore = responder.queue;
+        const item = responder.queue.queueItems[0];
+        await responder.txMined(item.request.identifier, item.nonce + 1);
+
+        expect(responder.queue).to.equal(queueBefore);
+        verify(transactionTrackerMock.addTx(anything(), anything())).once();
+    });
+});
```
Comment (bigspider) [inline test/src/responder/multiResponder.test.ts]: Perhaps add one or two tests for TransactionTracker.

File: test/src/responder/multiResponder.test.ts (hunk 2 above)
Comment (yahgwai) [reply test/src/responder/multiResponder.test.ts]: The plan in the long run will be to throw exceptions and handle these at the responder manager level, where we can retry or ask call other responders. For now I dont want errors to propogate to the watcher, so I but a large try/catch block here, but I could put one in the responder manager instead if you think that's better? My long term plan is to get this working in the anchor state paradigm, and then re-assess how these classes will interact.

File: test/src/responder/multiResponder.test.ts (hunk 3 above)
Comment (yahgwai) [reply test/src/responder/multiResponder.test.ts]: My plan is to remove the transaction tracker completley and replace it with an anchor state reducer

File: test/src/responder/multiResponder.test.ts (hunk 2 above)
Comment (bigspider) [reply test/src/responder/multiResponder.test.ts]: Fair enough, throwing errors in the Responders and handling them at the level of the ResponderManager seems appropriate to me.

**Pull request merged**